# 🐣 첫 표와 Series — Student

예측 → 작성 → 실행 → 확인 순서로 강의 코드를 연습한다.

> **🖥️ 환경 설정**
>
> - **로컬**: 배포 폴더에서 `uv sync` 후 `uv run jupyter lab`
>
> - **Colab**: 고정 CSV 세 파일을 `data/` 폴더에 업로드한 뒤 실행
>
> **⌨️ 단축키 안내**
>
> | 단축키 | 동작 |
> |--------|------|
> | `Shift + Enter` | 셀 실행 후 다음 셀 이동 |
> | `Esc → A` | 위에 셀 삽입 |
> | `Esc → B` | 아래에 셀 삽입 |
> | `Esc → DD` | 셀 삭제 |


In [1]:
# 🖥️ 환경 설정
# - 로컬: 터미널에서 `uv sync` 실행 (pyproject.toml + uv.lock 동봉)
# - Colab: 고정 CSV를 data/ 폴더에 올린 뒤 !pip install pandas -q

import pandas as pd


> **이 문법이 처음이면**
>
> 리스트는 순서, 딕셔너리는 이름과 값을 함께 담는다. Series는 둘 다에서 만들 수 있다.
>
> 더 볼 사람을 위한 자습 자료는 다음 사이클에 추가됩니다.


## 110-1 — FinanceDataReader 라이브러리 불러오기

**문제**

다음 한 줄을 그대로 입력하되 맨 앞에 `#`을 붙여 주석으로만 남기세요(실행하지 않습니다): `import FinanceDataReader as fdr`

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §1 강의노트](../notes/FD-02-110__fdr.md)

In [2]:
# 📌 §110 cell_id=110:1 실습
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
import FinanceDataReader as fdr
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


## 110-2 — KOSPI 상장 종목 목록 불러오기

**문제**

다음 한 줄을 그대로 입력하되 맨 앞에 `#`을 붙여 주석으로만 남기세요(실행하지 않습니다): `fdr.StockListing('KOSPI')`

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §2 강의노트](../notes/FD-02-110__fdr.md)

In [5]:
fdr.DataReader('005930','2026-01-01','2026-06-30',index_col="Date",parse_dates=True)

TypeError: DataReader() got an unexpected keyword argument 'index_col'

In [6]:
# 📌 §110 cell_id=110:2 실습
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
import pandas as pd
kospi=pd.read_csv('data/kospi_listing.csv',dtype={'Code':str})
kospi.head(3)
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


,Code,Name,Market,Sector,Marcap
0,005930,삼성전자,KOSPI,전기전자,1464492791304000
1,000660,SK하이닉스,KOSPI,전기전자,1178284184745000
2,005935,삼성전자우,KOSPI,전기전자,148839858156500


## 110-3 — KOSPI 종목표 CSV 읽고 앞 3행 확인

**문제**

`data/kospi_listing.csv`를 pandas로 읽어 `kospi`에 저장하세요. `Code`는 문자열로 읽어 `005930`, `000660`처럼 앞자리 0을 보존하고 `kospi.head(3)`을 출력하세요.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §2 강의노트](../notes/FD-02-110__fdr.md)

In [2]:
# 📌 §110 cell_id=110:3 실습
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
import pandas as pd
kospi=pd.read_csv('data/kospi_listing.csv',dtype={'Code':str})
kospi.head(3)
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


,Code,Name,Market,Sector,Marcap
0,005930,삼성전자,KOSPI,전기전자,1464492791304000
1,000660,SK하이닉스,KOSPI,전기전자,1178284184745000
2,005935,삼성전자우,KOSPI,전기전자,148839858156500


## 110-4 — 삼성전자 가격 데이터 불러오기

**문제**

다음 한 줄을 그대로 입력하되 맨 앞에 `#`을 붙여 주석으로만 남기세요(실행하지 않습니다): `fdr.DataReader('005930', '2026-01-02', '2026-06-30')`

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §3 강의노트](../notes/FD-02-110__fdr.md)

In [7]:
# 📌 §110 cell_id=110:4 실습
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.

#fdr.DataReader('005930', '2026-01-02', '2026-06-30')

samsung=pd.read_csv('data/samsung_005930.csv',index_col="Date",parse_dates=True)
samsung.shape
samsung.head(3)
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


,Open,High,Low,Close,Volume,Change
Date,,,,,,
2026-01-02,61900,62400,61400,62000,40702881,0.012900
2026-01-05,61800,62100,61600,61800,12673757,-0.003226
2026-01-06,62100,62500,61500,61600,38557870,-0.003236


## 110-5 — 삼성전자 가격표의 크기 확인

**문제**

`data/samsung_005930.csv`를 `index_col='Date', parse_dates=True`로 읽어 `samsung`에 저장하고 `samsung.shape`를 출력하세요. 결과는 (120, 6)입니다.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §3 강의노트](../notes/FD-02-110__fdr.md)

In [ ]:
# 📌 §110 cell_id=110:5 실습
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.

# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


## 110-6 — 삼성전자 가격표 앞 3행 확인

**문제**

앞 문제(110-5)에서 만든 `samsung`의 `samsung.head(3)`을 출력해 열 구성(`Open`, `High`, `Low`, `Close`, `Volume`, `Change`)을 확인합니다.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §3 강의노트](../notes/FD-02-110__fdr.md)

In [ ]:
# 📌 §110 cell_id=110:6 실습
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.

# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


## 110-7 — dtype 없이 읽을 때 종목코드 확인

**문제**

`data/kospi_listing.csv`를 `dtype` 없이 읽어 `wrong`에 저장하고 `wrong['Code'].head(3)`을 출력하세요. 앞자리 0이 사라진 정수(5930, 660, 5935)로 보이는지 확인합니다.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §4 강의노트](../notes/FD-02-110__fdr.md)

In [ ]:
# 📌 §110 cell_id=110:7 실습
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.

# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


## 200-1 — prices 리스트에서 세 번째 가격 선택

**문제**

`prices = [50000, 55000, 60000, 58000, 62000]`를 만들고 `prices[2]`를 출력하세요. 결과는 60000입니다.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §1 강의노트](../notes/FD-02-200__series.md)

In [7]:
# 📌 §200 cell_id=200:1 실습
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
prices=[50000,55000,60000,58000,62000]
print(prices[2])
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


60000


## 200-2 — prices 리스트를 기본 인덱스 Series로 변환

**문제**

앞 셀의 `prices`를 `pd.Series(prices)`로 변환해 `s`에 저장하고 `print(s)`로 기본 정수 인덱스(0~4)와 값을 확인하세요.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §2 강의노트](../notes/FD-02-200__series.md)

In [8]:
# 📌 §200 cell_id=200:2 🔀 번호표만 있는 것과 이름표가 붙은 것
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
s=pd.Series(prices)
print(s)
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


0    50000
1    55000
2    60000
3    58000
4    62000
dtype: int64


## 200-3 — 월 이름으로 Series 값 선택

**문제**

앞 셀의 `prices = [50000, 55000, 60000, 58000, 62000]`를 값으로 사용하고 인덱스를 `['1월', '2월', '3월', '4월', '5월']`로 지정해 `s`를 만드세요. `s['3월']`을 출력해 3월 가격 60,000을 선택하세요.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §2 강의노트](../notes/FD-02-200__series.md)

In [9]:
# 📌 §200 cell_id=200:3 🔀 번호표만 있는 것과 이름표가 붙은 것
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
s=pd.Series(prices,index=['1월','2월','3월','4월','5월'])
print(s['3월'])
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


60000


## 200-4 — 종목코드 딕셔너리를 Series로 변환

**문제**

`prices_dict = {'005930': 50000, '035420': 380000, '000660': 150000}`을 `pd.Series(prices_dict)`로 변환해 `s2`에 저장하고 `print(s2)`로 종목코드가 인덱스가 되는지 확인합니다.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §2 강의노트](../notes/FD-02-200__series.md)

In [11]:
# 📌 §200 cell_id=200:4 🔀 번호표만 있는 것과 이름표가 붙은 것
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
prices_dict={'005930': 50000,'035420':380000,'000660':150000}
s2=pd.Series(prices_dict)
print(s2)
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


005930     50000
035420    380000
000660    150000
dtype: int64


## 200-5 — Series 전체에 10% 상승 적용

**문제**

`s = pd.Series(prices, index=['1월', '2월', '3월', '4월', '5월'])`(200-3과 동일)를 다시 만들고 `s * 1.1`을 출력해 모든 가격에 10% 상승 시나리오를 한 번에 계산하세요.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §3 강의노트](../notes/FD-02-200__series.md)

In [12]:
# 📌 §200 cell_id=200:5 🔀 번호표만 있는 것과 이름표가 붙은 것
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
print(s2*1.1)
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


005930     55000.0
035420    418000.0
000660    165000.0
dtype: float64


## 200-6 — 매수가 대비 월별 수익률 계산

**문제**

앞 셀의 월별 가격 Series `s`를 사용하세요. 1월 매수가를 뜻하는 `buy_price = 50000`을 설정하고 `(s - buy_price) / buy_price`를 출력해 각 월의 매수가 대비 수익률을 계산하세요.

**작성 위치**: 바로 아래 코드 셀에 위 문제의 코드를 입력하세요. instructor와 student는 같은 문제를 공유합니다.

[→ §3 강의노트](../notes/FD-02-200__series.md)

In [ ]:
# 📌 §200 cell_id=200:6 🔀 번호표만 있는 것과 이름표가 붙은 것
# 코드를 작성해주세요 👇
# 과제: 강의노트의 입력과 같은 결과가 나오도록 한 단계씩 작성한다.
# 힌트: 바로 위의 설명과 고정 CSV의 열 이름을 다시 확인한다.
buy_price=50000
print((s-buy_price)/buy_price)
print(s.pct_change(buy_price))
# 예상 출력:
# 실행 후 강의노트의 결과와 비교하세요.


1월    0.00
2월    0.10
3월    0.20
4월    0.16
5월    0.24
dtype: float64
